app/tests/data_factory.py

```diff
@@ -132,7 +132,7 @@ def __init__(self, sample, required_only=False, **kwargs):
         try:
             sample = self.samples_dict[sample]
         except KeyError as error:
-            raise KeyError(f'Unknown {self.sample_name} sample: {error}')
+            raise KeyError(f'Unknown {self.sample_name} sample: {error}') from None
```

Comment:
Is `from None` correct here, or could/should it be `from error`?

Reply:
Yes it is correct, using `from error` would  give two backtrace, one from the original exception and one from the second. As the original exception doesn't provide more information as the one we re-raise (original exception message is part of the new one), we can omit it, this would make the reading of the exception easier.